### PR TITLE
Fix macro expanding, remove incorrect nightly check

### PR DESCRIPTION
**Pull request by puuuuh**
Fix macro expanding, remove incorrect nightly check in rustic-format-region, update syntax and idents in rustic-macro-expansion-mode

### CODE REVIEW

File: rustic-lsp.el [hunk 1]
```diff
@@ -152,18 +162,13 @@ with `lsp-rust-switch-server'."
       (let ((inhibit-read-only t))
         (erase-buffer)
         ;; wrap expanded macro in a main function so we can run rustfmt
-        (insert "fn main()")
+        (insert "fn main() {")
```
Comment (brotzeit) [inline rustic-lsp.el:165]: I guess they have changed the output.

File: rustic-rustfmt.el [hunk 2]
```diff
@@ -274,6 +294,20 @@ were issues when using stdin for formatting."
                                :buffer (current-buffer)
                                :stdin (buffer-string)))
 
+(defun rustic-format-macro-buffer ()
+  "Format the current buffer with macro output using rustfmt.
+
+Provide optional argument NO-STDIN for `rustic-before-save-hook' since there
+were issues when using stdin for formatting."
```
Comment (brotzeit) [inline rustic-rustfmt.el:301]: This sentence doesn't fit here.

File: rustic-lsp.el (hunk 1 above)
Comment (puuuuh) [reply rustic-lsp.el:165]: Attribute macroses can expand to one or more statements and cant be formatted with old method

File: rustic-rustfmt.el [hunk 3]
```diff
@@ -261,10 +281,7 @@ This operation requires a nightly version of rustfmt.
 
 ;;;###autoload
 (defun rustic-format-buffer ()
-  "Format the current buffer using rustfmt.
-
-Provide optional argument NO-STDIN for `rustic-before-save-hook' since there
-were issues when using stdin for formatting."
```
Comment (brotzeit) [inline rustic-rustfmt.el:267]: It seems the line wasn't removed when we changed this function in the past. Thanks for fixing it.

File: rustic-lsp.el (hunk 1 above)
Comment (brotzeit) [reply rustic-lsp.el:165]: I see.

**Pull request merged**